atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -863,6 +863,13 @@ public String toString() {
                         RowColumnRangeExtractor extractor = new RowColumnRangeExtractor(metricsManager);
                         extractor.extractResults(ImmutableList.of(row), results, startTs);
                         RowColumnRangeExtractor.RowColumnRangeResult decoded = extractor.getRowColumnRangeResult();
+                        if (decoded.getResults().isEmpty()) {
+                            // All returned results were at ts > startTs
+                            // Start the next page from the last column seen to ensure forward progress
+                            byte[] lastColumn = extractor.getLastColumnForRow(row).get();
```

Comment:
`.get()` here should be safe since we are collecting results only for a single row, and we have already confirmed that results were returned - there must be at least one column for which results were returned.